dav-controller/api/routers/age_verification.py

```diff
@@ -101,18 +102,25 @@ async def get_dav_request(pid: str, db: Database = Depends(get_db)):
             for key, value in revealed_attr_value_dict.items():
                 resp_incl_revealed_attibs[key] = value["raw"]
 
-        metadata = auth_session.metadata or {}
+        metadata = auth_session.metadata or {} 
         metadata["revealed_attributes"] = resp_incl_revealed_attibs
 
-        # Needs to be made flexible for different proof requests
         response = AgeVerificationModelRead(
             status=auth_session.proof_status,
             id=str(auth_session.id),
             notify_endpoint=auth_session.notify_endpoint,
             metadata=metadata,
         )
-        # Testing
-        logger.error(f" --- {str(response)}")
+        logger.debug(f"Generated response: {response}")
+        # Remove revealed attributes if not retained
+        if not auth_session.retain_attributes:
+            update_metadata = copy.deepcopy(metadata)
+            if "revealed_attributes" in update_metadata:
+                del update_metadata["revealed_attributes"]
+                logger.info(f"Attributes not retained for session {auth_session.id}")
+
+            update_data = AuthSessionPatch(metadata=update_metadata, pres_exch_id=auth_session.pres_exch_id)
+            await AuthSessionCRUD(db).patch(str(auth_session.id), update_data)
```

Comment:
This updates the auth session and removes the data after the first query: in an edge scenario where - for some reason - the endpoint is not called after the proof is sent to the verifier, we would still have attributes stored in the database, right?

I am wondering if we should/could instead update the `acapy_handler` code to ignore the attributes when updating the authsession instead, to prevent them from entering the database in the first place.